From 3d32e5266f8128181c48c6a016cd6958d33aeaa6 Mon Sep 17 00:00:00 2001
From: Marek Waszkiewicz <mwaszkiewicz@anaconda.com>
Date: Wed, 20 Dec 2023 14:49:02 +0100
Subject: [PATCH] change loadlibrary path

---
 mclbn256/mclbn256.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/mclbn256/mclbn256.py b/mclbn256/mclbn256.py
index 93a04f1..84dedd4 100644
--- a/mclbn256/mclbn256.py
+++ b/mclbn256/mclbn256.py
@@ -2,6 +2,8 @@ from ctypes import Structure, c_uint64, cdll, c_char_p, create_string_buffer
 from hashlib import blake2b, sha256
 import platform
 import pkg_resources
+import os
+import sys
 
 def load_library(path_name):
     if platform.system() == 'Windows':
@@ -10,7 +12,7 @@ def load_library(path_name):
         lib_ext = ".dylib"
     else:
         lib_ext = ".so"
-    return cdll.LoadLibrary(pkg_resources.resource_filename('mclbn256', path_name+lib_ext))
+    return cdll.LoadLibrary(path_name + lib_ext)
 
 #
 # Define constants needed to replace the C headers
@@ -344,7 +346,11 @@ class lib:
 def __init_lib():
     global lib
     # _ = load_library("lib/libmcl")
-    lib = load_library("libmclbn256")
+    if sys.platform == 'win32':
+        lib = load_library(os.path.join(os.environ['CONDA_PREFIX'], 'Library', 'bin', "mclbn256"))
+    else:
+        lib = load_library(os.path.join(os.environ['CONDA_PREFIX'], 'lib', "libmclbn256"))
+
     if lib.mclBn_init(mclBn_CurveFp254BNb, MCLBN_COMPILED_TIME_VAR): print("Failed to load MCl's BN254 binary.")
 __init_lib()
 
-- 
2.25.1

